### PR TITLE
chore: rename maxToolsUsePerRun to maxStepsPerRun

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

`maxToolsUsePerRun` limits the number of steps an assistant run and not the total number of tools (so the name is wrong).

This is step 1 out of 2 to rename the column: I'm introducing a new column `maxStepsPerRun` + a backfill migration for it.

Next PR will be removing `maxToolsUsePerRun`

## Risk

As always such migrations are risky. 
The change has been properly tested.

## Deploy Plan

1. run the first migration (`migration_41` to add the new field as nullable)
2. deploy the code
3. run the backfill script
4. run `migration_42` (make the field no nullable)
5. [NEXT PR] remove all usage of `maxToolsUsePerRun` in the code
6. [NEXT PR] drop the `maxToolsUsePerRun` column

### CODE REVIEW

**Review by spolu — APPROVED** [review]
LGTM

File: front/pages/api/w/[wId]/assistant/agent_configurations/[aId]/index.ts [hunk 1]
```diff
@@ -97,35 +97,33 @@ async function handler(
         });
       }
 
-      const maxToolsUsePerRun =
-        bodyValidation.right.assistant.maxToolsUsePerRun;
+      const maxStepsPerRun = bodyValidation.right.assistant.maxStepsPerRun;
```
Comment (PopDaph) [inline front/pages/api/w/[wId]/assistant/agent_configurations/[aId]/index.ts:100]: We need to send both previous and new fields on the AssistantBuilder form to not error for all people that are in the AssistantBuilder when you deploy this new code :/

Comment (fontanierh) [reply front/pages/api/w/[wId]/assistant/agent_configurations/[aId]/index.ts:100]: Indeed :/

**Review by PopDaph — APPROVED** [review]
🙏🏻

**Pull request merged**